dali/pipeline/data/tensor_vector.h

```diff
@@ -122,40 +122,40 @@ class DLL_PUBLIC TensorVector {
    */
   std::vector<size_t> _chunks_capacity() const;
 
-  TensorListShape<> shape() const;
+  const TensorListShape<> &shape() const;
```

Comment:
```suggestion
  const TensorListShape<> &shape() const &;
```
Just in case...

Also, you can add an overload with `&&` qualifier which returns the shape by value - or simply delete it.